### ISSUE TITLE
FaceLivenessDetector orientation

### ISSUE THREAD

**Issue by gabriel-amorim** [aws-amplify/amplify-ui-android#45]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui-android/issues?q=is%3Aissue+).

### Which UI component is this feature-request for?

Liveness

### Please include any relevant guides or documentation you're referencing

https://github.com/aws-amplify/amplify-ui-android/blob/main/liveness/src/main/java/com/amplifyframework/ui/liveness/ui/FaceLivenessDetector.kt

### Please describe your feature-request in detail.

The screen orientation lock function on line 102 of the FaceLivenessDetector.kt source could lock it in the app's current orientation rather than just in portrait mode. This way, the liveness screen could run in both orientations.

Currently if you run a LivenessSession in an application that runs in landscape orientation, the SDK forces the orientation to portrait mode causing the application to crash.

### Code Snippet

```java
// Put your code below this line.

```

### Additional information and screenshots

![image](https://github.com/aws-amplify/amplify-ui-android/assets/45129357/407da942-57d2-457d-a83d-6ef050d17057)

**Comment by tylerjroach** [comment]
@gabriel-amorim Please provide a stacktrace of the crash. The FaceLivenessDetector UI is only designed to work in portrait at this moment.

**Comment by gabriel-amorim** [comment]
@tylerjroach 

It happens when you pass 'disableStartView = true' to FaceLivenessDetector.

```
FATAL EXCEPTION: main
Process: com.postomirian.sam, PID: 25039
java.lang.IllegalArgumentException: Trying to create LifecycleCamera with destroyed lifecycle.
at androidx.camera.lifecycle.LifecycleCameraRepository.createLifecycleCamera(LifecycleCameraRepository.java:107)
at androidx.camera.lifecycle.ProcessCameraProvider.bindToLifecycle(ProcessCameraProvider.java:503)
at androidx.camera.lifecycle.ProcessCameraProvider.bindToLifecycle(ProcessCameraProvider.java:363)
at com.amplifyframework.ui.liveness.camera.LivenessCoordinator$1.invokeSuspend(LivenessCoordinator.kt:141)
at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
at android.os.Handler.handleCallback(Handler.java:942)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loopOnce(Looper.java:226)
at android.os.Looper.loop(Looper.java:313)
at android.app.ActivityThread.main(ActivityThread.java:8757)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@7b42b80, Dispatchers.Main]

```

**Comment by tjleing** [comment]
I was able to reproduce this issue and find the root cause:

When `disableStartView = true` is set the relevant code looks like this:

```
// in compose
LockPortraitOrientation()
ChallengeView()

fun ChallengeView() {
    val lifecycleOwner = LocalLifecycleOwner.current
    coordinator = LivenessCoordinator(lifecycleOwner)
}

class LivenessCoordinator(lifecycleOwner) {
    init {
        MainScope().launch {
            getCameraProvider(context).apply {
                    bindToLifecycle(lifecycleOwner)
            }
        }
    }
}
```

The main issue is that we create two `LivenessCoordinator`s, each with their own `LifecycleOwner`,  but then initializing the first coordinator is delayed enough that the `LifecycleOwner` is destroyed before it can be used.

SImple solution: just check if the owner is valid before using it.  Which works... the first time you run the liveness check; if you try to restart the check it'll either enter an infinite loop of switching from landscape to portrait or just crash with this same error.  I'll update this issue if I make any progress.

**Comment by tjleing** [comment]
This change was released with Amplify UI v1.1.2, @gabriel-amorim please see if that has resolved your issue!
